File: IntroJava/src/PrimeNumber.java
/*
Listing 5.15 PrimeNumber.java
This section presents a program that displays the first 50 prime numbers in 5 lines,
each containing 10 numbers.
 */

import java.util.Scanner;

public class PrimeNumber
{
    public static void main(String[] args)
    {
        final int NUMBER_OF_PRIME = 50;
        int count = 0; // number of prime
        final int NUMBER_OF_PRIME_PER_LINE = 10;
        int number = 2;  // number to be tested for prime

        while (count < NUMBER_OF_PRIME)
        {
            boolean isPrime = true;
            for (int divisor = 2; divisor <= number / 2; divisor++)
            {
                if (number % divisor == 0)
                {
                    isPrime = false;
                    break;
                }
            }

            if (isPrime)
            {
                count ++;

                if (count % NUMBER_OF_PRIME_PER_LINE == 0)
                    System.out.println(number);
                else
                    System.out.print(number + " ");
            }

            number ++;
        }
    }
}

File: IntroPython/PrimeNumber.py
# LISTING 5.13 PrimeNumber.py

NUMBER_OF_PRIMES = 50
NUMBER_OF_PRIMES_PER_LINE = 10
count = 0  # Count the number of prime numbers
number = 2  # A number to be tested for primeness

print("The first 50 prime numbers are ")

while count < NUMBER_OF_PRIMES:
    isPrime = True  # Is the current number prime?

    divisor = 2
    while divisor <= number / 2:
        if number % divisor == 0:
            isPrime = False
            break

        divisor += 1

    if isPrime:
        count += 1
        print(format(number, "5d"), end="")
        if count % NUMBER_OF_PRIMES_PER_LINE == 0:
            print()

    number += 1
